EIPS/eip-7508.md

```diff
@@ -0,0 +1,301 @@
+---
+eip: 7508
+title: Ethereum Entity Component System
+description: A minimal Entity Component System framework built on Ethereum
+author: Rickey (@HelloRickey)
+discussions-to: https://ethereum-magicians.org/t/a-new-proposal-of-entity-component-system/15665
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-09-05
+---
+
+## Abstract
+
+This proposal defines a minimal Entity Component System, which will help developers clean and easy to use the Entity Component System framework on Ethereum.
+
+## Motivation   
+
+ECS is a design pattern that improves code reusability by separating data from behavior. It is often used in game development.A minimal ECS consists of   
+**Entity**: a unique identifier.   
+**Component**: a reusable data container attached to an entity.   
+**System**: the logic for operating entity components.   
+**World**: a container for an entity component system.   
+This proposal uses smart contracts to implement an easy-to-use minimal ECS, eliminates unnecessary complexity, and makes some functional improvements that are consistent with contract interaction behavior. You can combine components and systems easily and freely. And make it upgradable through state management. It will help you quickly launch your full-chain game or application.
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Interfaces
+
+**IWorld.sol**
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+interface IWorld {
+    /**
+     * Create a new entity.
+     * @dev The entity MUST be assigned a unique Id.
+     * If the state of the entity is true, it means it is available, and if it is false, it means it is not available.
+     * When the state of the entity is false, you cannot add or remove components for the entity.
+     * @param _entityState is the state of the entity.
+     * @return New entity id.
+     */
+    function createEntity(bool _entityState) external returns (uint256);
+
+    /**
+     * Does the entity exist in the world.
+     * @param _entityId is the Id of the entity.
+     * @return true exists, false does not exist.
+     */
+    function entityExists(uint256 _entityId) external view returns (bool);
+
+    /**
+     * Get the total number of entities in the world.
+     * @return The total number of entities.
+     */
+    function getEntityCount() external view returns (uint256);
+
+    /**
+     * Set the state of an entity.
+     * @dev Entity MUST exist.
+     * @param _entityId is the Id of the entity.
+     * @param _entityState is the state of the entity.
+     */
+    function setEntityState(uint256 _entityId, bool _entityState) external;
```

Comment:
The comments here don't give enough context for what a boolean entity state is supposed to represent?